src/components/TextField/index.d.ts

```diff
@@ -0,0 +1,22 @@
+import { FunctionComponent, ReactNode } from "react";
+import { Theme } from "styled-system";
+import { CommonProps } from "../../utils/common";
+
+export interface TextFieldProps extends CommonProps {
+  "aria-label": string;
+  // TODO: what is this? - what type should it be
+  as: any;
```

Comment:
same as any `as` - it should be `ReactType`. https://styled-components.com/docs/api#as-polymorphic-prop